.github/workflows/test-crates.yml

```diff
@@ -1,81 +1,73 @@
-# This workflow facilitates the individual building of Rust crates present in the repository.
-# 1. A matrix is generated dynamically to identify each crate in the repository.
-# 2. This matrix is checked for validity.
-# 3. Each identified crate undergoes three build processes:
-#    - With no features.
-#    - With the default features.
-#    - With all the features enabled.
-# 4. In case of build failures outside of pull requests, an issue is either opened or updated
-#    in the repository to report the failure.
-# Throughout the workflow, various setup steps ensure the correct environment and tools are present.
-name: Build crates individually
-
-# Ensures that only one workflow task will run at a time. Previous builds, if
-# already in process, will get cancelled. Only the latest commit will be allowed
-# to run, cancelling any workflows in between
-concurrency:
-  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
-  cancel-in-progress: true
+name: Test Crate Build
 
 on:
-  workflow_dispatch:
-  push:
-    branches:
-      - main
+  pull_request:
+    branches: [main]
     paths:
-      # production code and test code
       - "**/*.rs"
-      # dependencies
       - "**/Cargo.toml"
       - "**/Cargo.lock"
-      # configuration files
-      - ".cargo/config.toml"
+      - .cargo/config.toml
       - "**/clippy.toml"
-      # workflow definitions
-      - ".github/workflows/ci-build-crates.yml"
-  pull_request:
+      - .github/workflows/test-crate-build.yml
+
+  push:
+    branches: [main]
     paths:
-      # production code and test code
       - "**/*.rs"
-      # dependencies
       - "**/Cargo.toml"
       - "**/Cargo.lock"
-      # configuration files
-      - ".cargo/config.toml"
+      - .cargo/config.toml
       - "**/clippy.toml"
-      # workflow definitions
-      - ".github/workflows/ci-build-crates.yml"
+      - .github/workflows/test-crate-build.yml
+
+  workflow_dispatch:
+
+# Ensures that only one workflow task will run at a time. Previous builds, if
+# already in process, will get cancelled. Only the latest commit will be allowed
+# to run, cancelling any workflows in between
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
+permissions:
+  contents: read
+  pull-requests: read
 
 env:
-  CARGO_INCREMENTAL: ${{ vars.CARGO_INCREMENTAL }}
-  RUST_LOG: ${{ vars.RUST_LOG }}
-  RUST_BACKTRACE: ${{ vars.RUST_BACKTRACE }}
-  RUST_LIB_BACKTRACE: ${{ vars.RUST_LIB_BACKTRACE }}
-  COLORBT_SHOW_HIDDEN: ${{ vars.COLORBT_SHOW_HIDDEN }}
+  CARGO_TERM_COLOR: always
+  RUST_BACKTRACE: 1
+  RUST_LIB_BACKTRACE: 1
+  RUST_LOG: info
+  COLORBT_SHOW_HIDDEN: 1
 
 jobs:
   matrix:
     name: Generate crates matrix
```

Comment:
I think this job could be greatly simplified by using `cargo hack build --workspace` (instead of manually building a crates matrix as it does currently) since it will build each crate separately if I understood it correctly. However feel free to do that in a separate PR if you prefer.

Reply:
I do agree. We've had it this way because if fail-fast and we have a better UX in the GitHub console, but we can discuss if this is not needed and simplify this.
<img width="636" height="565" alt="image" src="https://github.com/user-attachments/assets/be505e8b-20ec-4965-8262-904d8137240a" />

Reply:
I don't mind either way, but would slightly prefer to have one job as I agree the list of tests it too big.